src/Symfony/Component/Messenger/Middleware/SendMessageMiddleware.php

```diff
@@ -66,6 +70,12 @@ public function handle(Envelope $envelope, StackInterface $stack): Envelope
                 $this->logger->info('Sending message {class} with {alias} sender using {sender}', $context + ['alias' => $alias, 'sender' => \get_class($sender)]);
                 $envelope = $sender->send($envelope->with(new SentStamp(\get_class($sender), \is_string($alias) ? $alias : null)));
             }
+
+            if (false === $this->throwNoRouting
+                && 0 === iterator_count($this->sendersLocator->getSenders($envelope))
```

Comment:
We should not call the function ever. instead we should add a check that will tell if we entered the loop above, eg by setting $alias to null before the foreach, and checking if it's still null here.

Reply:
or better, by doing the same check against the `$sender` variable, which is already used like that below